### ISSUE TITLE
setting CDPATH breaks libxsmm build

### ISSUE THREAD

**Issue by hajgato** [easybuilders/easybuild-framework#2944]
If you have `CDPATH` environmental variable, it breaks the build of `libxsmm`. (version 1.8.3 confirmed, second run, when shared objects created from static libraries). It is probably caused by `patsubst` make function or automake or something else, as issuing `cd` command might generate an output if `CDPATH` is set. It makes sense to unset `CDPATH` as it might cause problems in other places as well.

**Comment by boegel** [comment]
Confirmed here, simply setting `$CDPATH` to something trivial like `.:/tmp` is enough to cause trouble (e.g. break the installation of `libxsmm-1.8.3-foss-2018a.eb`).

We should simply always unset `$CDPATH` if it is defined, there's no benefit that `eb` can have from having it set...

**Comment by boegel** [comment]
Some more context:

```
$ cd foo
cd: no such file or directory: foo

$ mkdir /tmp/foo
$ export CDPATH=.:/tmp
$ cd foo
/tmp/foo

$ pwd
/tmp/foo
```

The output that `cd foo` produces (similar to `cd -`) can mess up some things...

**Comment by boegel** [comment]
fixed in #2970
